Compares with files example (#1918)
saucelabs/sauce-docs: docs/api-testing/use-cases/use-drive.md

      mode: json
  expression: myUserDetails
```

## Comparing the Response to a File

> The last example considered is when you need to compare the response from a call to a "template" to verify that it is precisely the same. Also, in this case, you must save the template in the Vault Drive and compare it with the response.

Reviewer: ```suggestion
For this last example consider a case where you need to compare the response from a call to a "template" to verify that they are the same.  You can  save the template in the Vault Drive and compare it with the response.
```

### Step 1: Uploading the File in Vault Drive

> The first thing is uploading the file into the Vault. You can follow the [same steps as in the first example](#step-1-uploading-the-file-in-vault-drive), but this time you will use the following payload, saved as _user.json_:

Reviewer: ```suggestion
First, upload file in the Vault. You can follow the [same steps as in the first example](#step-1-uploading-the-file-in-vault-drive), but this time you will use the following payload, saved as _user.json_:
```

#### Parsing the File

The next step is parsing the file to let the system know the type of file you are working with.

Add the **Parse** component:

- The **Variable** must match the name you entered as a Variable in the previous step.
- The **Adapter** must match the type of file you have uploaded. In our example, it's a .json file.

<img src={useBaseUrl('/img/api-testing/vault-use-cases/parsejson.png')} alt="Parse the file"/>

#### Adding the Request

Add the request to the endpoint you want to compare the file with.

1. Add the **POST** component:
   - URL - for example `https://eo51qycg3neycjd.m.pipedream.net/user`
   - Variable - for example `payload`
   - Mode - for example `json`
1. **Save Changes**.

<img src={useBaseUrl('/img/api-testing/vault-use-cases/post_user.png')} alt="Add the POST request"/>

#### Comparing the File to the Response

Next, you can compare the file to the response using the **Assert compares** component.

1. Add the **Assert Compares** component.

   - Expression 1 - for example `sample_payload`
   - Expression 2 - for example `payload`
   - Mode - for example `structure`
   - Strict - for example `No`

>    In **Expression 1** and **Expression 2**, you have to enter the variable names you have entered for the file and the call.

Reviewer: ```suggestion
   In **Expression 1** and **Expression 2**, enter the variable names you have entered for the file and the call.
```

>    Mode `structure` means you will compare the JSON structure of the template and the response

Author: ```suggestion
   Mode `structure` means you will compare the JSON structure of the template and the response.
```

>    Mode `structure` means you will compare the structure of the two json.

Reviewer: ```suggestion
   Mode `structure` means you will compare the JSON structure of the template and the response
```

   Set strict to `Yes` if you want to include data types in the comparison.

1. Add the **Assert Compares** component and set **Mode** to `values` to compare the values in the file and the ones in the call.

   - Expression 1 - for example `sample_payload`
   - Expression 2 - for example `payload`
   - Mode - for example `value`
   - Strict - for example `No`

   <img src={useBaseUrl('/img/api-testing/vault-use-cases/compares.png')} alt="Add the POST request"/>

Now you can proceed with your test.

The final result of the test looks like this:

```yaml
- id: set
  var: sample_payload
  mode: object
  object: DS.loadTextFile('user.json')
- id: parse
  var: sample_payload
  adapter: json
- id: post
  children: []
  url: https://eo51qycg3neycjd.m.pipedream.net/user
  var: payload
  mode: json
- id: assert-compares
  expression1: payload
  expression2: sample_payload
  mode: structure
  strict: "true"
- id: assert-compares
  expression1: sample_payload
  expression2: payload
  mode: values
  strict: "true"
```

> :::Note Text Mode

Reviewer: ```suggestion
:::note Text Mode
```

> When you need to compare a string such as an Authorization token, or an ID (not necessarily from a file) you 1text1 **Mode**.

Author: ```suggestion
When you need to compare a string such as an Authorization token, or an ID (not necessarily from a file) you can choose `text` **Mode**.
```

> When you need to compare a string such as an Authorization token, or an ID (not necessarily from a file) the correct **Mode** to set up is `text`.

Reviewer: ```suggestion
When you need to compare a string such as an Authorization token, or an ID (not necessarily from a file) you 1text1 **Mode**.
```
